cli/k8s/kubectl.go

```diff
@@ -104,6 +111,59 @@ func (kctl *kubectl) UrlFor(namespace string, extraPathStartingWithSlash string)
 	return generateKubernetesApiBaseUrlFor(schemeHostAndPort, namespace, extraPathStartingWithSlash)
 }
 
+func (kctl *kubectl) SelfCheck() ([]healthcheck.CheckResult, error) {
+	results := make([]healthcheck.CheckResult, 0)
```

Comment:
what do you think about pre-populating all 3 checks here? i'm thinking about the case where the implementation of this function changes to where it exits early, and does not append all its checks to the array. it could show an incomplete picture of health status depending on earlier checks failing.

Reply:
I've interpreted this as 40aa6d3, am I understanding your comment right?

Reply:
i guess i had in mind that if the function exited early it would return all checks, whether they ran or not, but i'm realizing that introduces an entirely new "did not run" state. i think what you've just proposed is better.